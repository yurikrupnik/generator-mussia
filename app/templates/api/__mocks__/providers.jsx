import React from 'react';
import PropTypes from 'prop-types';

const A = ({ children }) => (
    <div>
        {children}
    </div>
);

A.propTypes = {
    children: PropTypes.element.isRequired
};

const B = ({ children }) => (
    <div>
        {children}
    </div>
);

B.propTypes = {
    children: PropTypes.element.isRequired
};

export default [
    A,
    B
];
