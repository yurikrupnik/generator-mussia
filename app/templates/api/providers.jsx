import users from './users/provider';
import projects from './projects/provider';

export default [
    users,
    projects
];
