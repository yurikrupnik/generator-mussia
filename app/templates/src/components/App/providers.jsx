import { createElement } from 'react';
import PropTypes from 'prop-types';

const Providers = ({ children, providers }) => providers
    .reduceRight((acc, c) => createElement(c, {}, acc), children);

Providers.propTypes = {
    children: PropTypes.element.isRequired,
    providers: PropTypes.arrayOf(PropTypes.func).isRequired
};

export default Providers;
