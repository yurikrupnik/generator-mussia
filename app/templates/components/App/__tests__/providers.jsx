import React from 'react';
import Providers from '../providers';
import providers from '../../../api/providers';

// jest.mock('../../../api/providers');

const {
    describe,
    it,
    expect,
    shallow,
} = global;

describe('providers', () => {
    it('renders three <Providers /> components', () => {
        const wrapper = shallow(
            <Providers providers={providers}>
                <div>
                    asd
                </div>
            </Providers>
        );
        expect(wrapper.length).toBe(1);
        expect(wrapper).toMatchSnapshot();
    });
});
