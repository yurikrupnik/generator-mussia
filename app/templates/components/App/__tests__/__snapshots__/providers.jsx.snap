// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`providers renders three <Providers /> components 1`] = `
<UsersProvider>
  <ProjectsProvider>
    <div>
      asd
    </div>
  </ProjectsProvider>
</UsersProvider>
`;
